### PR TITLE
Adds OnClearFromRecentService example

### PR DESCRIPTION
**Pull request by atreat**
Example to receive callback when the app is removed from the system recents menu
Adds Notification logic to OnClearFromRecentService
Adds to README about OnClearFromRecentService

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
 # Drive Science Android Demo App
 
-This app is a demonstration of how you might integrate with the Android Trip Tracker SDK. For further documentation on the integration process, please refer to the [Android docs](https://github.com/Root-App/drive-science-docs/blob/master/android.md).
\ No newline at end of file
+This app is a demonstration of how you might integrate with the Android Trip Tracker SDK. For further documentation on the integration process, please refer to the [Android docs](https://github.com/Root-App/drive-science-docs/blob/master/android.md).
```
Comment (frankmassi-root) [inline README.md:3]: Not a change for this PR, but why does this not live in [readme like our ios docs](https://developer.root-enterprise.com/docs/ios) do?

The repo you linked to even says it is deprecated as a resource.

image

File: app/src/main/java/com/example/drivescience/OnClearFromRecentService.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    private void createNotification() {
+        createNotificationChannel();
+
+        Intent intent = new Intent(this, MainActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, intent, 0);
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(this, "1234")
```
Comment (frankmassi-root) [inline app/src/main/java/com/example/drivescience/OnClearFromRecentService.java:49]: What is "1234"? That _feels_ like dummy test data, but I'm guessing it's something to do with android internals?

Comment (frankmassi-root) [reply app/src/main/java/com/example/drivescience/OnClearFromRecentService.java:49]: Or should this be using the `NOTIFICATION_ID` you defined up above?

**Review by frankmassi-root — COMMENTED** [review]
Not fully sure how android notifications work, but it looks like you maybe forgot to use the constant you defined for notification id in one place?

File: app/src/main/java/com/example/drivescience/OnClearFromRecentService.java (hunk 2 above)
Comment (atreat) [reply app/src/main/java/com/example/drivescience/OnClearFromRecentService.java]: This needed to match the channel ID actually. Updated :+1:

**Review by frankmassi-root — APPROVED** [review]
I still stand by my comment about the docs link, but I think we need to actually do more work than would be in scope for this PR.

**Review by camneu-root — APPROVED** [review]

**Comment by atreat** [comment]
@frankmassi-root agree on the link to the docs. Will update in a follow up PR

**Pull request merged**